charmcraft/utils.py

```diff
@@ -151,16 +163,12 @@ def get_os_platform(filepath=pathlib.Path("/etc/os-release")):
     if system == "Linux":
         os_release = {}
         try:
-            with filepath.open("r", encoding='utf-8') as f:
-                for line in f:
-                    if "=" in line:
-                        key, value = line.rstrip().split("=", 1)
-                        if value[0] == value[-1] == '"':
-                            value = value[1:-1]
-                        os_release[key] = value
+            with filepath.open("rt", encoding='utf-8') as fh:
+                lines = [x.strip() for x in fh.readlines() if x.strip()]
         except FileNotFoundError:
             logger.debug("Unable to locate 'os-release' file, using default values")
         finally:
+            os_release = parse_keqv_list(lines)
```

Comment:
Just know that this would break if a commented line were in the os-release file.  Not that I'm aware of any cases where it happens, but it's allowed in the spec.

Reply:
Indeed. I added that and other cases to the tests, and improved the code.